modules/siddhi-extensions/markov-models/src/test/java/org/wso2/siddhi/extension/markov/test/util/SiddhiTestHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied. See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.wso2.siddhi.extension.markov.test.util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SiddhiTestHelper {
```

Comment:
@ruwanta Thanks for bringing up this point about CountDownLatch. Yes, I agree with your point about Thread.sleep() looking bad here. If we can improve the code, IMHO, we should improve even if it is just a Test. Thinking further on your suggestion and thinking about how the Siddhi callbacks operate, I think CountDownLatch can be used here efficiently without any issue. 

@ashensw , shall we consider the CountDownLatch for this particular extension and see. Since a failing test can block the whole test suite if simply CountDownLatch.await() is called, we would always have to use CountDownLatch.await(long timeout) version AFAIU. If the CountDownLatch is passed down to the helper method, similar to how the Atomic variable is passed down, and if CountDownLatch.countDown() is called within the SiddhiCallback, I think it will work without any issue. The only downside is that the CountDownLatch will have to be instantiated for each test case (as opposed to resetting the Atomic counter in our current test cases). But the approach suggested by @ruwanta feels much cleaner. WDYT?